src/app/auth/auth.guard.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  RouterStateSnapshot,
+} from '@angular/router';
+import { AuthService } from './auth.service';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+
+  constructor(
+    private router: Router,
+    private authService: AuthService,
+  ) { }
+
+  canActivate(_: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+    return this.checkAuth(state.url);
+  }
+
+  private checkAuth(url: string) {
+    if (this.authService.isAuthenticated) {
+      // logged in so return true
+      return true;
+    }
+
+    // not logged in so redirect to login page with the return url and return false
```

Comment:
nit: Personally don't think the comments in this method are necessary, but if keeping them then I suggest rewording slightly.
```suggestion
    // not logged in so set the return url, redirect to the login page, and return false
```

Reply:
I've removed the comments.